src/core/unit.c

```diff
@@ -107,6 +107,8 @@ Unit *unit_new(Manager *m, size_t size) {
         u->ipv6_allow_map_fd = -1;
         u->ipv4_deny_map_fd = -1;
         u->ipv6_deny_map_fd = -1;
+        u->ip_allow_any = false;
+        u->ip_deny_any = false;
```

Comment:
hmm, do we really have to define caching variables for this?

I think it would be nicer to just add a small helper to ip-address-access.[ch] maybe called:

`bool ip_address_access_item_is_any(IPAddressAccessItem *first);`

that function would return true if the list consists exactly of the the two items necessary for "any". It is possible to determine this in O(1) time, hence doing so should be quick and much simpler, as we avoid caching and double-storing stuff.

Reply:
I cached them like this because
a. the access maps were not stored in the Unit and
b. to avoid having to loop through the slices and figure out if a parent had "any" defined somewhere in [bpf_firewall_compile](https://github.com/systemd/systemd/pull/12745/files/a37b8302826bbb0670d13d77cbfa57b65ce89039#diff-75070c719bae7a214e5b2429a7d087d8R579). The loop was already done at [bpf_firewall_prepare_access_maps](https://github.com/systemd/systemd/blob/master/src/core/bpf-firewall.c#L388-L396) so I figured caching the result from there and reusing it later was worth the extra variable.

Not sure which one is better. I could also pass the result through a local variable instead of saving it in Unit.

Reply:
our current assumption is that the depth of the slices isn't too bad (i.e. below 10), i don't think we need to cache that, it sounds like premature optimization. In 99% of cases the depth is probably not going to exceed 4 i guess, which should be quick enough